src/lib/model/contract.ts

```diff
@@ -120,14 +126,9 @@ export const useContractData = (
 
   const { data: contractBalances, isLoading: isContractBalancesLoading } =
     useQuery(
-      ["query", "contractBalances", contractAddress, chainId],
-      async () =>
-        queryContractBalances(
-          currentChainRecord?.name,
-          currentChainRecord?.chain.chain_id,
-          contractAddress
-        ),
-      { enabled: !!currentChainRecord && !!contractAddress, retry: false }
+      ["query", "contractBalances", balancesApiRoute, contractAddress],
+      async () => queryContractBalances(balancesApiRoute, contractAddress),
+      { enabled: !!balancesApiRoute && !!contractAddress, retry: false }
```

Comment:
Leave it as !!contractAddress because we assume that balancesApiRoute must be available.